tesseract/tesseract_scene_graph/include/tesseract_scene_graph/srdf_model.h

```diff
@@ -0,0 +1,182 @@
+  using ChainGroups = std::unordered_map<std::string, std::vector<std::pair<std::string, std::string>>>;
+  using JointGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using LinkGroups = std::unordered_map<std::string, std::vector<std::string>>;
+  using GroupOPWKinematics = std::unordered_map<std::string, OPWKinematicParameters>;
+
+  SRDFModel() = default;
+  virtual ~SRDFModel() = default;
+  SRDFModel(const SRDFModel&) = default;
+  SRDFModel& operator=(const SRDFModel&) = default;
+  SRDFModel(SRDFModel&&) = default;
+  SRDFModel& operator=(SRDFModel&&) = default;
+
+  /** @brief Load Model from TiXMLElement */
+  bool initXml(const tesseract_scene_graph::SceneGraph& scene_graph, tinyxml2::XMLElement* srdf_xml);
+
+  /** @brief Load Model from TiXMLDocument */
+  bool initXml(const tesseract_scene_graph::SceneGraph& scene_graph, tinyxml2::XMLDocument* srdf_xml);
+
+  /** @brief Load Model given a filename */
+  bool initFile(const tesseract_scene_graph::SceneGraph& scene_graph, const std::string& filename);
+
+  /** @brief Load Model from a XML-string */
+  bool initString(const tesseract_scene_graph::SceneGraph& scene_graph, const std::string& xmlstring);
+
+  /** @brief Save the model to a file */
+  bool saveToFile(const std::string& file_path) const;
+
+  /** @brief Get the name of this model */
+  const std::string& getName() const;
+  std::string& getName();
+
+  /** @brief Get the version number {major, minor, patch} */
+  const std::array<int, 3>& getVersion() const;
+
+  /** @brief Get the allowed collision matrix */
+  const AllowedCollisionMatrix& getAllowedCollisionMatrix() const;
+  AllowedCollisionMatrix& getAllowedCollisionMatrix();
+
+  /** @brief Get the map of chain groups defined for this model */
+  const ChainGroups& getChainGroups() const;
+  ChainGroups& getChainGroups();
+
+  /** @brief Get the map of joint groups defined for this model */
+  const JointGroups& getJointGroups() const;
+  JointGroups& getJointGroups();
+
+  /** @brief Get the map of groups defined for this model */
+  const LinkGroups& getLinkGroups() const;
+  LinkGroups& getLinkGroups();
+
+  /** @brief Get the map of group tool center points (TCP) defined for this model */
+  const GroupTCPs& getGroupTCPs() const;
+  GroupTCPs& getGroupTCPs();
+
+  /** @brief Get the map of group states defined for this model */
+  const GroupStates& getGroupStates() const;
+  GroupStates& getGroupStates();
+
+  /** @brief Get the map of group opw kinematics defined for this model */
+  const GroupOPWKinematics& getGroupOPWKinematics() const;
+  GroupOPWKinematics& getGroupOPWKinematics();
+
+  /** @brief Clear the model */
+  void clear();
+
+private:
+  /**
+   * @brief Load groups from srdf xml element
+   * @param scene_graph The tesseract scene graph
+   * @param srdf_xml The xml element to parse
+   */
+  void loadGroups(const tesseract_scene_graph::SceneGraph& scene_graph, tinyxml2::XMLElement* srdf_xml);
```

Comment:
I assume the xml element can't be const for some reason?

Reply:
Fixed.